### PR TITLE
Cribl phase2

### PR DESCRIPTION
**Pull request by kgeller**
## Proposed commit message

This is the second phase of the Cribl integration. It goes alongside a custom kibana UI that is being added [here](https://github.com/elastic/kibana/pull/176439).

This PR 1) updates the documentation, 2) adds the configuration variable for routing (which is a json string) and 3) bumps the required stack version that the UI changes will (hopefully) be added.

## Checklist

- [X] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [X] I have verified that all data streams collect metrics or logs.
- [X] I have added an entry to my package's `changelog.yml` file.
- [X] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## Related issues

- Requires https://github.com/elastic/kibana/pull/176439

### CODE REVIEW

File: packages/cribl/_dev/build/docs/README.md [hunk 1]
```diff
@@ -0,0 +1,47 @@
+# Cribl
```
Comment (jamiehynds) [inline packages/cribl/_dev/build/docs/README.md:1]: @alaudazzi would you mind reviewing this readme for a Cribl integration we're working on?

File: packages/cribl/data_stream/logs/elasticsearch/ingest_pipeline/default.yml [hunk 2]
```diff
@@ -3,12 +3,18 @@ description: Pipeline for rerouting log streams from Cribl.
...
   - append:
       field: tags
       value: 
         - cribl
       allow_duplicates: false
+  - rename:
+      field: _raw
+      target_field: message
+      ignore_missing: true
+  - pipeline:
```
Comment (kgeller) [inline packages/cribl/data_stream/logs/elasticsearch/ingest_pipeline/default.yml:16]: this pipeline gets created/updated via kibana 

https://github.com/elastic/kibana/blob/main/x-pack/plugins/security_solution/server/security_integrations/handlers/put_cribl_routing_pipeline.ts#L31

https://github.com/elastic/kibana/blob/main/x-pack/plugins/security_solution/server/lib/security_integrations/cribl/util/pipeline_builder.ts

File: packages/cribl/_dev/build/docs/README.md [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+        See [Cribl Data Onboarding](https://docs.cribl.io/stream/data-onboarding/) for more information on configuring sources.
+
+    2. Configure the Cribl integration in Kibana
+
+        Note: The Cribl integration does not require Elastic Agent, but a policy must be configured when setting up the Cribl integration.
+
+4. Configure an Elastic destination in Cribl
+
+    Cribl offers two options for sending data to Elastic, the Elastic Cloud ouput for cloud environments, and the Elasticsearch output for self-managed. Consult the or the [Cribl Elastic Cloud documentation](https://docs.cribl.io/stream/destinations-elastic-cloud/) or [Cribl Elasticsearch documentation](https://docs.cribl.io/stream/destinations-elastic/) for more details on how to configure.
```
Comment (kcreddy) [inline packages/cribl/_dev/build/docs/README.md:35]:
```suggestion
    Cribl offers two options for sending data to Elastic, the Elastic Cloud ouput for cloud environments, and the Elasticsearch output for self-managed. Consult [Cribl Elastic Cloud documentation](https://docs.cribl.io/stream/destinations-elastic-cloud/) or [Cribl Elasticsearch documentation](https://docs.cribl.io/stream/destinations-elastic/) for more details on how to configure.
```

File: packages/cribl/_dev/build/docs/README.md [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+
+    **Destination settings**
+
+    1. Set **Cloud Id** for the Cloud destination or **Bulk API URLs** for the Elasticaearch destination to point to your Elastic cluster.
+
+    2. Set **Index or Data Stream** to `logs-cribl-default`.
+
+    3. **API key** should be a Base64 encoded Elastic API key, which can be created in Kibana by following the instructions under API Keys. If you are using an API key with “Restrict privileges”, be sure to review the Indices privileges to provide at least "auto_configure" & "write" permissions for the logs-* index, which you will be using for these Fleet integration data streams.
```
Comment (kcreddy) [inline packages/cribl/_dev/build/docs/README.md:43]:
```suggestion
    3. **API key** should be a Base64 encoded Elastic API key, which can be created in Kibana by following the instructions under **Management** > **Stack Management** > **Security** > **API Keys**. If you are using an API key with “Restrict privileges”, be sure to review the Indices privileges to provide at least "auto_configure" & "write" permissions for the logs-* index, which you will be using for these Fleet integration data streams.
```

**Review by kcreddy — COMMENTED** [review]
Few nits

File: packages/cribl/_dev/build/docs/README.md [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+
+    1. Set **Cloud Id** for the Cloud destination or **Bulk API URLs** for the Elasticaearch destination to point to your Elastic cluster.
+
+    2. Set **Index or Data Stream** to `logs-cribl-default`.
+
+    3. **API key** should be a Base64 encoded Elastic API key, which can be created in Kibana by following the instructions under **Management** > **Stack Management** > **Security** > **API Keys**. If you are using an API key with “Restrict privileges”, be sure to review the Indices privileges to provide at least "auto_configure" & "write" permissions for the logs-* index, which you will be using for these Fleet integration data streams.
+
+
+
+
```
Comment (efd6) [inline packages/cribl/_dev/build/docs/README.md:47]:
```suggestion
```

File: packages/cribl/data_stream/logs/elasticsearch/ingest_pipeline/default.yml (hunk 2 above)
Comment (efd6) [reply packages/cribl/data_stream/logs/elasticsearch/ingest_pipeline/default.yml:16]: Can this be documented in the pipeline here with a link to elastic/kibana#176439?

**Review by kcreddy — APPROVED** [review]
LGTM 👍🏼 
Once Dan's comments are resolved, its good to merge.

**Review by efd6 — APPROVED** [review]

File: packages/cribl/_dev/build/docs/README.md (hunk 1 above)
Comment (alaudazzi) [reply packages/cribl/_dev/build/docs/README.md:1]: Sorry for the late reply, I must have skipped the notification. Reviewing it now.

File: packages/cribl/_dev/build/docs/README.md [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+        Note: The Cribl integration does not require Elastic Agent, but a policy must be configured when setting up the Cribl integration.
+
+4. Configure an Elastic destination in Cribl
+
+    Cribl offers two options for sending data to Elastic, the Elastic Cloud output for cloud environments, and the Elasticsearch output for self-managed. Consult [Cribl Elastic Cloud documentation](https://docs.cribl.io/stream/destinations-elastic-cloud/) or [Cribl Elasticsearch documentation](https://docs.cribl.io/stream/destinations-elastic/) for more details on how to configure.
+
+    **Destination settings**
+
+    1. Set **Cloud Id** for the Cloud destination or **Bulk API URLs** for the Elasticaearch destination to point to your Elastic cluster.
```
Comment (alaudazzi) [inline packages/cribl/_dev/build/docs/README.md:39]:
```suggestion
    1. Set **Cloud Id** for the Cloud destination or **Bulk API URLs** for the Elasticsearch destination to point to your Elastic cluster.
```

File: packages/cribl/_dev/build/docs/README.md [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+
+    2. Set **Index or Data Stream** to `logs-cribl-default`.
+
+    3. **API key** should be a Base64 encoded Elastic API key, which can be created in Kibana by following the instructions under **Management** > **Stack Management** > **Security** > **API Keys**. If you are using an API key with “Restrict privileges”, be sure to review the Indices privileges to provide at least "auto_configure" & "write" permissions for the logs-* index, which you will be using for these Fleet integration data streams.
```
Comment (alaudazzi) [inline packages/cribl/_dev/build/docs/README.md:43]:
```suggestion
    3. **API key** should be a Base64 encoded Elastic API key, which you can create in Kibana by following the instructions under **Management** > **Stack Management** > **Security** > **API Keys**. If you are using an API key with “Restrict privileges”, be sure to review the Indices privileges to provide at least "auto_configure" and "write" permissions for the logs-* index, which you will be using for these Fleet integration data streams.
```
prefer active over passive voice wherever possible

**Review by alaudazzi — APPROVED** [review]
I left a few minor editing suggestions, otherwise LGTM.

**Comment by alaudazzi** [comment]
@kgeller 
Maybe you can handle it in a separate PR, but it would be good to apply the [documentation guidelines](https://github.com/elastic/integrations/blob/main/docs/documentation_guidelines.md).

**Comment by kgeller** [comment]
> @kgeller Maybe you can handle it in a separate PR, but it would be good to apply the [documentation guidelines](https://github.com/elastic/integrations/blob/main/docs/documentation_guidelines.md).

@alaudazzi Which aspects of the guidelines are you referring? 

I covered only the overview and the setup instructions since that's all I felt applied, but am open to adding more. This integration is different in that it doesn't really do much other than take events and route them to other datastreams from other integrations. So it's like a pass through mechanism and can be used in front of any of the other integrations.

**Pull request merged**